### DISCUSSION TITLE
Running on demand - Intune Remediation Script + Blacklist URL

### DISCUSSION THREAD

**Discussion by colourofsound** [Romanitho/Winget-AutoUpdate#612] [Q&A]
I'm a bit wary of running anything as a scheduled task on a machine, simply because if I need it to stop, getting rid of it via Intune scripts is never a sure thing.

I really like using Remediations in Intune for this instead. So I'm looking at installing WAU and then triggering it with remediations.

So, on to my questions: 

- Can WAU be triggered on demand? I assume just running 'Winget-Upgrade.ps1' would do it; but I want to ensure it runs silently.
- If WAU runs as both System and User, is there ever a risk of the user update run prompting for Admin creds?
- I'm having trouble understanding how to use a Blacklist as a URL. My blacklist exists in an Azure storage blob container which is publicly accessible by URL. However, the install script can't find the list, whether its public, private with SAS, all the special characters dealt with...doesn't make sense

Thanks in advance! Using WAU in combination with [Winget Wrapper](https://github.com/SorenLundt/WinGet-Wrapper) is such a huge time-saver :)

**Comment by colourofsound** [comment]
Just answered my own question by looking at the scheduled task:

powershell.exe -NoProfile -ExecutionPolicy Bypass -File "C:\ProgramData\Winget-AutoUpdate\winget-upgrade.ps1"

So I think this could be run as a remediation in Intune with the trigger/detection being if winget returns anything when running 'winget upgrade'

**Reply by Romanitho** [reply]
You can install WAU without any trigger and I guess you can use remediation scripts to start WAU "on demand" yes.
